### PR TITLE
Update PR template to ask for release type

### PR DESCRIPTION
**Pull request by isra-fel**
## Description

There have been some cases where we merged PRs that should target a preview to the `main` branch and they went in the stable releases. This could impact the stability of the stable releases.

The reason behind is that we expect the service teams to proactively tell us it's a preview, but that doesn't always happen.
So this PR added a question that service team must answer before we merge.

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
 ## Checklist
 
+- Is this targeting a **general** release / a **preview** release / an **engineering build** of Azure PowerShell?
+    - `Private Preview`
```
Comment (NoriZC) [inline .github/PULL_REQUEST_TEMPLATE.md:10]: Could you explain more about the Private Preview case? 
- Why it is listed singularly?
- Where should the pr go if it is a private preview?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -6,6 +6,8 @@
 
 ## Checklist
 
+- Is this targeting a **general** release / a **preview** release / an **engineering build** of Azure PowerShell?
```
Comment (NoriZC) [inline .github/PULL_REQUEST_TEMPLATE.md:9]:
```suggestion
- Please confirm your target, is it a **general** release / a **preview** release / an **engineering build** of Azure PowerShell?
- `<put your answer here>`
```

**Review by NoriZC — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (isra-fel) [reply .github/PULL_REQUEST_TEMPLATE.md]: To explain the context, I updated "CONTRIBUTING.md" and link it here.

**Review by isra-fel — COMMENTED** [review]
- Remove section about design review
- Consider remove "SHOULD select appropriate branch. Cmdlets from Autorest.PowerShell should go to [generation](https://github.com/Azure/azure-powershell/tree/generation) branch.“
- Consider remove "SHOULD make the title of PR clear and informative, and in the present imperative tense."
- Consider remove "SHOULD NOT introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/ea5cafad75de07da4c77cd10eaa95d5b181826a4/blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version."

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -90,6 +90,25 @@ completed checklist.
...
 
+#### Target Release Types
+
+Here are all types of releases of **Azure PowerShell**. Note that the type of release isn't necessarily aligned to the state of the swagger or service feature.
+
+- **General** release
+  - This PR targets an [official general release](https://github.com/Azure/azure-powershell/milestones) of Azure PowerShell.
+  - Target version is `x.y.z`, no suffix.
+  - Target branch should be `main` or `generation` depending on the development approach.
+- **Preview** release
```
Comment (dolauli) [inline CONTRIBUTING.md:101]: For preview release, shall we include the case for 0.x.y releases?

Comment (isra-fel) [reply CONTRIBUTING.md:102]: Good point. Updated.

**Review by dolauli — APPROVED** [review]

**Pull request merged**